### DISCUSSION TITLE
Web server / client support?

### DISCUSSION THREAD

**Discussion by zackify** [microsoft/devicescript#440] [General]
With the pico w, is there any chance of getting a simple web server like micro python has?

I know there is the local cloud stuff that is being worked on, but I would love being able to make a simple server or make http requests to things.

**Comment by pelikhan** [comment]
Related to MQTT (https://github.com/microsoft/devicescript/issues/395), supporting different networking setups.

**Comment by pelikhan** [comment]
We are looking into this for the ESP32.

**Comment by pelikhan** [comment]
We have added "fetch" and sockets in the latest release (v2.10.*)

**Comment by zackify** [comment]
Wow that was fast! That’s so awesome thanks

**Comment by zackify** [comment]
Any way for me to know when this will be in the docs? I didn’t see any notes on release either

**Comment by pelikhan** [comment]
The docs are not quite there yet but the fetch implementation should give you a good idea of how to use sockets.

https://microsoft.github.io/devicescript/developer/net

**Comment by pelikhan** [comment]
https://github.com/microsoft/devicescript/blob/main/packages/net/src/fetch.ts

**Comment by zackify** [comment]
Thanks so much that’s perfect on the client side, just need to learn how I can start a http server next!

Edit: nice I’m reading your socket code. Very helpful

**Reply by pelikhan** [reply]
(moved to discussion) keep us up to date on the server impl

**Reply by zackify** [reply]
Also, will http 2 support be added to fetch? May be worth looking into

**Reply by pelikhan** [reply]
I think http2 support will have to be a community contribution. which feature in particular?

**Reply by arxenix** [reply]
is tcp/tls server support coming @pelikhan ?

**Reply by pelikhan** [reply]
We are not planning to work on server support but we welcome contributions.
